app/controllers/user_passwords_controller.rb

```diff
@@ -26,4 +28,13 @@ def create
   def set_admin_redirect
     session["spree_user_return_to"] = params[:return_to] if params[:return_to]
   end
+
+  def user_unconfirmed?
+    user = Spree::User.find_by_email(params[:spree_user][:email])
+    if user && !user.confirmed?
+      render json: { error: t('email_unconfirmed') }, status: :unauthorized
+    end
+
+    user && !user.confirmed?
+  end
```

Comment:
I find this method a bit confusing due to the fact that is named as a boolean method but it also renders a JSON, which is a bit unexpected. What about extracting the render out of this method?